gemma-core/src/main/java/ubic/gemma/persistence/persister/AuditablePersister.java

```diff
@@ -0,0 +1,45 @@
+package ubic.gemma.persistence.persister;
+
+import org.hibernate.FlushMode;
+import org.hibernate.SessionFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.transaction.annotation.Transactional;
+import ubic.gemma.model.common.Auditable;
+import ubic.gemma.model.common.auditAndSecurity.AuditTrail;
+
+/**
+ * Use this as a base class for {@link Auditable} entities.
+ * @param <T>
+ */
+public abstract class AuditablePersister<T extends Auditable> extends AbstractPersister<T> {
```

Comment:
This is where the magic happens for `Auditable` entities. By extending this, we make sure that an audit trail is always created on persist, the subclass implements persistAuditable with specific logic.